certificateshop-logic/src/main/java/com/epam/esm/dao/impl/jpa/JpaOrderDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+        this.orderExtractor = orderExtractor;
+    }
+
+    /**
+     * Saves {@link Order} in the database.
+     *
+     * @param order is the {@link Order} to save.
+     */
+    @Override
+    public void save(Order order) {
+        entityManager
+                .createNativeQuery(INSERT_ENTITY_SQL)
+                .setParameter(1, order.getUser().getId())
+                .setParameter(2, order.getCreateDate())
+                .setParameter(3, order.getName())
+                .executeUpdate();
+    }
+
+    /**
+     * Finds all {@link Order} entity in the database.
+     *
+     * @return List of the {@link Order} objects.
+     */
+    @Override
+    public List<Order> findAll() {
+        Query query = entityManager.createNativeQuery(FIND_ALL_ENTITIES_SQL);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        return getEntities(result);
+    }
+
+    /**
+     * Finds {@link Optional <Order>} in the database by the id of the {@link Order}.
+     *
+     * @param id is the {@link long} to find.
+     * @return {@link Optional<Order>}.
+     */
+    @Override
+    public Optional<Order> findById(long id) {
+        Query query = entityManager.createNativeQuery(FIND_ENTITY_BY_ID_SQL).setParameter(1, id);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        List<Order> orders = getEntities(result);
+        return orders.stream().findFirst();
+    }
+
+    /**
+     * Updates the {@link Order}.
+     *
+     * @param order is the value of the {@link Order} to update.
+     */
+    @Override
+    public void update(Order order) {
+
+    }
+
+    /**
+     * Removes the {@link Order} object from the database.
+     *
+     * @param id is the value of the {@link long} to find.
+     */
+    @Override
+    public void delete(long id) {
+
+    }
+
+    /**
+     * Finds {@link Optional<Order>} in the database by the id of the {@link Order}.
+     *
+     * @param name is the {@link String} to find.
+     * @return {@link Optional<Order>}.
+     */
+    @Override
+    public Optional<Order> findByName(String name) {
+        Query query = entityManager.createNativeQuery(FIND_ENTITY_BY_NAME_SQL).setParameter(1, name);
+        List<Object[]> resultList = query.getResultList();
+        List<List<Object>> result = convertListOfArrayToListOfLists(resultList);
+        List<Order> users = getEntities(result);
+        if (users.isEmpty()) {
+            return Optional.empty();
+        }
+        return users.stream().findFirst();
+    }
+
+    private List<List<Object>> convertListOfArrayToListOfLists(List<Object[]> resultList) {
+        List<List<Object>> result = new ArrayList();
+        for (Object[] objects : resultList) {
+            List<Object> listOfObjects = new ArrayList<>(Arrays.asList(objects));
```

Comment:
It's better to name this variable as `objectList`

Reply:
do you mean 'result' or 'listOfObjects'?